app/app/calculations/punching.py

```diff
@@ -0,0 +1,224 @@
+        sr_max = 0.75 * d 	# promieniowy rozstaw strzemion
+        fywd_ef = max((250 + 0.25 * d / mm) * MPa, fywd)
+        asw_req = (ved1 - 0.75 * vrdc) * sr_max / (1.5 * d) * u1 * d / (fywd_ef * sin(alfa))
+        asw_1bar = pi * dsw**2 / 4
+        nsw_req = asw_req / asw_1bar
+
+        # Zbrojenie minimalne
+        ro_sw_min = 0.08 * (fck / MPa)**0.5 * MPa / fyk
+        st_max = 2.0 * d 	# styczny rozstw strzemion
+        asw_min = ro_sw_min * sr_max * st_max / (1.5 * sin(alfa) + cos(alfa))	# czy aby poprawna intepretacja pola jedengo ramienia strzemion??????
+        nsw_min = asw_min / asw_1bar
+
+        # Przyjecie zbrojenia
+        nsw_prov = ceil(max(nsw_req, nsw_min))
+        asw_prov = nsw_prov * asw_1bar
+
+        # Nośność na ścinanie przy przebiciu ze zbrojeniem na przebicie (vRdcs)
+        vrdcs = 0.75 * vrdc + 1.5 * d / sr_max * asw_prov * fywd_ef / (u1 * d) * sin(alfa)
+
+        if vrdcs >= ved1:
+            ERRORS.append('Nośność na ścinanie przy przebiciu zachowana (vEd1 <= vRdcs).')
+            print('Nośność na ścinanie przy przebiciu zachowana (vEd1 <= vRdcs).')
+        else:   # elif vrdcs < ved1:
+            ERRORS.append('Nośność na ścinanie przekroczona (vEd1 > vRdcs)!')
+            print('Nośność na ścinanie przekroczona (vEd1 > vRdcs)!')
+
+        # Rozmieszczenie zbrojenia
+            # Co najmniej 2 obwody strzemin
+            # Max rozstaw obwodow 0.75d
+            # Max rozstaw strzemion w obwodzie 2d (1.5d dla u1)
+  
+        u_out = beta * ved / (vrdc * d)
+
+        a_out = compute_a_out1(support, u_out, cx, cy, lx, ly) # TODO: nigdzie nie zostało użyte
+
+    # TODO: do zrobienia powyżej, natomiast poniżej do sprawdzenia, czy jest ok
+    vrdc = asw_prov / cm / cm
+    vrdmax = asx / cm / cm
+    return vrdc, vrdmax, ERRORS
+
+
+def compute_u00(support_id, cx, cy, d):
+    # TODO: zrobić coś z tymi komentarzami, w bazie dodatkowo:
+    # TODO: ściana - naroże (3) i ściana - koniec (4)
+    if support_id == 0:
+        return 2 * (cx + cy)
+    elif support_id == 1:   # Poprawic, poki co jest roboczo !!!
+        return min(2 * cx + cy, cy + 3 * d)
+    #elif support_id == 'słup krawędziowy X':
+    #    return min(2 * cx + cy, cy + 3 * d)
+    #elif support_id == 'słup krawędziowy Y':
+    #    return min(cx + 2 * cy, cx + 3 * d)
+    elif support_id == 2:
+        return min(cx + cy, 3 * d)
+    else:
+        ERRORS.append('Błąd w określeniu położenia słupa.')
+        return None
+
+
+def compute_u11(support_id, cx, cy, d, lx, ly):
+    # TODO: jak wyżej
+    if support_id == 0:
+        return 2 * cx + 2 * cy + 4 * pi * d
+    if support_id == 1:   # Poprawic, poki co jest roboczo !!!
+        return 2 * cx + 1 * cy + 2 * pi * d + 2 * lx
+    # elif support_id == 'słup krawędziowy X':
+    #     return 2 * cx + 1 * cy + 2 * pi * d + 2 * lx
+    # elif support_id == 'słup krawędziowy Y':
+    #     return 1 * cx + 2 * cy + 2 * pi * d + 2 * ly
```

Comment:
Koniecznie trzeba to poprawić, im szybciej, tym lepiej. W pozostałych funkcjach to samo.